### DISCUSSION TITLE
Select not selected after reload - @persist plugin

### DISCUSSION THREAD

**Discussion by n0ah01** [alpinejs/alpine#3142] [7. Bugs (archived)] [answered]
## Context
I'm building a config page with lots of repeater-groups with inputs/selects, that should be persistant across reloads.

## Problem
Inputs retain the set value correctly. The same goes for selects, but the selected option is not correctly "reselected" on a reload.

## Example code
```
<div x-data="{
  testInput: $persist(''),
  testSave: $persist(''),
  testData: ['One', 'Two', 'Three'],
}">
  <input type="text" x-model="testInput">
  <select class="select select-sm" x-model="testSave">
    <option value="" disabled>Test</option>
    <template x-for="ele in testData">
      <option x-text="ele"></option>
    </template>
  </select>
  <div>
    Persist Value: <span x-text="testSave"></span>
  </div>
</div>
```

**Comment by ekwoka** [comment]
To clarify, you mean that the value is in the data component, but does not set the `select` element to have that same value on mount?

I think there could be some conflict with this. When there is a set value in the component data, but also a `selected` option, which one should win at mount?

I think right now they don't do anything. They just stay out of sync.

**Comment by SimoTod** [comment, marked as answer]
The script needs `<option x-text="ele" :selected="ele == testSave"></option>`
x-model gets processed before x-for in that script so the option list doesn't contain testData and the browser can't select the right option

**Reply by n0ah01** [reply]
Thank you very much!
I thought that something like this is missing, but while testing I could not get it to work...
